internal/kinds/capp/resourcemanagers/dnsrecord.go

```diff
@@ -0,0 +1,229 @@
+				return err
+			}
+			if capp.Status.RouteStatus.DomainMappingObjectStatus.URL != nil {
+				if err := r.handlePreviousDNSRecords(capp, resourceManager, dnsRecordFromCapp.Name); err != nil {
+					return fmt.Errorf("failed to delete previous Certificates: %w", err)
+				}
+			}
+		} else {
+			return fmt.Errorf("failed to get DNSRecord %q: %w", dnsRecordFromCapp.Name, err)
+		}
+	}
+
+	return r.updateDNSRecord(aRecordSet, dnsRecordFromCapp, resourceManager)
+}
+
+// createDNSRecord creates a new DNSRecord and emits an event.
+func (r DNSRecordManager) createDNSRecord(capp cappv1alpha1.Capp, dnsRecordFromCapp dnsrecordv1alpha1.CNAMERecord, resourceManager rclient.ResourceManagerClient) error {
+	if err := resourceManager.CreateResource(&dnsRecordFromCapp); err != nil {
+		r.EventRecorder.Event(&capp, corev1.EventTypeWarning, eventCappDNSRecordCreationFailed,
+			fmt.Sprintf("Failed to create DNSRecord %s", dnsRecordFromCapp.Name))
+
+		return err
+	}
+
+	r.EventRecorder.Event(&capp, corev1.EventTypeNormal, eventCappDNSRecordCreated,
+		fmt.Sprintf("Created DNSRecord %s", dnsRecordFromCapp.Name))
+
+	return nil
+}
+
+// updateDNSRecord checks if an update to the DNSRecord is necessary and performs the update to match desired state.
+func (r DNSRecordManager) updateDNSRecord(dnsRecord, dnsRecordFromCapp dnsrecordv1alpha1.CNAMERecord, resourceManager rclient.ResourceManagerClient) error {
+	if !reflect.DeepEqual(dnsRecord.Spec, dnsRecordFromCapp.Spec) {
+		dnsRecord.Spec = dnsRecordFromCapp.Spec
+		return resourceManager.UpdateResource(&dnsRecord)
+	}
+
+	return nil
+}
+
+// handlePreviousDNSRecords takes care of removing unneeded DNSRecord objects. If the new DNSRecord
+// is not yet available then return early and do not delete the previous Records.
+func (r DNSRecordManager) handlePreviousDNSRecords(capp cappv1alpha1.Capp, resourceManager rclient.ResourceManagerClient, name string) error {
+	available, err := utils.IsCNAMERecordAvailable(r.Ctx, r.K8sclient, name, capp.Namespace)
+	if err != nil {
+		return err
+	}
+
+	if !available {
+		return nil
+	}
+
+	dnsRecords, err := r.getPreviousDNSRecords(capp)
+	if err != nil {
+		return err
+	}
+
+	return r.deletePreviousARecordSets(dnsRecords, resourceManager, capp.Spec.RouteSpec.Hostname)
+}
+
+// getPreviousDNSRecords returns a list of all DNSRecord objects that are related to the given Capp.
+func (r DNSRecordManager) getPreviousDNSRecords(capp cappv1alpha1.Capp) (dnsrecordv1alpha1.CNAMERecordList, error) {
+	dnsRecords := dnsrecordv1alpha1.CNAMERecordList{}
+
+	nameRequirement, err := labels.NewRequirement(CappResourceKey, selection.Equals, []string{capp.Name})
+	if err != nil {
+		return dnsRecords, fmt.Errorf("unable to create label requirement for Capp: %w", err)
+	}
+
+	nsRequirement, err := labels.NewRequirement(cappNamespaceKey, selection.Equals, []string{capp.Namespace})
+	if err != nil {
+		return dnsRecords, fmt.Errorf("unable to create label requirement for Capp: %w", err)
+	}
```

Comment:
I think we should make util unction for labels which just accepts map of string string and returns labelSelector